### PR TITLE
libfido2: 1.5.0 -> 1.7.0

### PR DESCRIPTION
**Pull request by prusnak**
###### Motivation for this change

* https://developers.yubico.com/libfido2/Release_Notes.html

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by prusnak** [comment]
@GrahamcOfBorg build libfido2

**Comment by prusnak** [comment]
@GrahamcOfBorg build libfido2

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 120652` at 685e424c run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>479 packages marked as broken and skipped:</summary>
  <ul>
    <li>agdaPackages.iowa-stdlib</li>
    <li>aldor</li>
    <li>aqemu</li>
    <li>atom-beta</li>
    <li>autotrace</li>
    <li>bareos</li>
    <li>bonfire</li>
    <li>cassandra_2_1</li>
    <li>cassandra_2_2</li>
    <li>dockbarx</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>12 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/2652919bd175cda5ff3786f3d5bda85c">cargo</a></li>
    <li><a href="https://gist.github.com/3e9392860e60f492d6b1748530cb4008">git</a></li>
    <li><a href="https://gist.github.com/ef538f7e9778227a06b6b43667fecf1f">librsvg</a> (gnome3.librsvg)</li>
    <li><a href="https://gist.github.com/9373c78015f8e58258423740f3a7fcbd">iconnamingutils</a></li>
    <li><a href="https://gist.github.com/a9b8c58fb8469ec62f6823525de03b8d">libfido2</a></li>
    <li><a href="https://gist.github.com/6986d540ab1acab925f2079d4c3ad36d">openssh</a></li>
    <li><a href="https://gist.github.com/98cc7d167f5e780fc7d8e54076370877">python38Packages.GitPython</a></li>
    <li><a href="https://gist.github.com/174117b9ad7829cb04f79bf9eef81587">python38Packages.dateparser</a></li>
    <li><a href="https://gist.github.com/6ac0253d195e3a7350bfaecf1dc23e48">python38Packages.pytest-fixture-config</a></li>
    <li><a href="https://gist.github.com/791cf2367c1a7fafdf07ddf10c215158">python38Packages.setuptools-git</a></li>
    <li><a href="https://gist.github.com/e8ba8caf936de8927914182e8949509b">python39Packages.setuptools-git</a></li>
    <li><a href="https://gist.github.com/efe0e85103c9925f630d6092614ec97e">rdocker</a></li>
  </ul>
</details>
<details>
  <summary>5246 packages skipped due to time constraints:</summary>
  <ul>
    <li>AusweisApp2</li>
    <li>DisnixWebService</li>
    <li>Fabric (python38Packages.Fabric)</li>
    <li>MIDIVisualizer</li>
    <li>R</li>
    <li>Sylk</li>
    <li>_1password-gui</li>
    <li>abcl</li>
    <li>abiword</li>
    <li>accountsservice</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `fetchpatch`.
Near pkgs/development/libraries/libfido2/default.nix:3:3:
```
  |
3 | , fetchpatch
  |   ^
```
</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement.

**Comment by prusnak** [comment]
@GrahamcOfBorg build libfido2

**Comment by prusnak** [comment]
@GrahamcOfBorg build libfido2

**Review by mmilata — APPROVED** [review]
Did not test but diff LGTM.

File: pkgs/development/libraries/libfido2/default.nix [hunk 1]
```diff
@@ -7,30 +8,45 @@
...
 , udev
+, zlib
 }:
 
 stdenv.mkDerivation rec {
   pname = "libfido2";
-  version = "1.5.0";
+  version = "1.7.0";
   src = fetchurl {
     url = "https://developers.yubico.com/${pname}/Releases/${pname}-${version}.tar.gz";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libfido2/default.nix:18]: Any reason why we are not fetching from https://github.com/Yubico/libfido2/releases/tag/1.7.0 ?

Comment (prusnak) [reply pkgs/development/libraries/libfido2/default.nix:18]: Releases on https://developers.yubico.com/libfido2/Releases/ are signed

Comment (SuperSandro2000) [reply pkgs/development/libraries/libfido2/default.nix:18]: Are we checking that?

Comment (prusnak) [reply pkgs/development/libraries/libfido2/default.nix:18]: not automatically, but I do that manually

**Pull request merged**